src/core/server.ts

```diff
@@ -179,7 +179,8 @@ export class Server<C extends BaseContext> {
 
     debug('start:ApolloServerAllocation:start');
     this.graphQLServer = new ApolloServer({
-      context: (options: { req: Request }) => {
+      context: async (options: { req: Request }) => {
+        const consumerCtx = await contextGetter(options.req);
```

Comment:
Should the default `contextGetter` return a Promise above then?

Reply:
no need.  it will pass through.